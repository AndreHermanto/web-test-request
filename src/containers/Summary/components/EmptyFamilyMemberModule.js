import React, { Component } from 'react';
import { 
  Row, 
  Col,
  Glyphicon,
  Button 
} from 'react-bootstrap';
import { 
  SummaryBox,
  SummaryHeading,
  SummaryBreakLine,
  SummaryDetails
} from '../summaryStyled';

/**
* EmptyFamilyMemberModule summary - UI for summary page to display all family member details/clinical info data.
*/
class EmptyFamilyMemberModule extends Component
{
  constructor(props) {
    super(props);
    this.handleOnClick = this.handleOnClick.bind(this);
  }

  handleOnClick() {
    this.props.handleOnClick('step4');
    return 4;
  }

  render() {
    return(
      <SummaryBox>
        <SummaryHeading> 
          Family Member 
          <Button
            bsStyle="link"
            onClick={this.handleOnClick}
          >
            <Glyphicon glyph="pencil"/> Edit
          </Button>
        </SummaryHeading>
        <SummaryBreakLine/>
        <Row>
          <Col md={12}>
            <SummaryDetails> 
              No family member is included for this test.
            </SummaryDetails>
          </Col>
        </Row>
      </SummaryBox>
    )
  }
}

export default EmptyFamilyMemberModule;