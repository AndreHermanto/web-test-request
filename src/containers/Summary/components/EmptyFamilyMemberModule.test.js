import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import TestUtils from 'react-addons-test-utils';
import EmptyFamilyMemberModule from './EmptyFamilyMemberModule';

function getStep(value) {
  return value;
}
describe('EmptyFamilyMemberModule test', () => {
  const props = {
    handleOnClick: getStep
  };
    
  test('renders without crashing ', () => {
    const page = renderer.create(React.createElement(EmptyFamilyMemberModule, props)).toJSON();
    expect(page).toMatchSnapshot();
  });
  
  test('handleOnClick test', () => {
    var page = TestUtils.renderIntoDocument(
      <EmptyFamilyMemberModule
              handleOnClick={getStep}/>
    );
    expect(page.props.handleOnClick('step4')).toEqual('step4');
    expect(page.handleOnClick()).toEqual(4);
  });
});
