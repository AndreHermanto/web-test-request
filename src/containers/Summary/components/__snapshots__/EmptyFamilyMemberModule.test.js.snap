exports[`EmptyFamilyMemberModule test renders without crashing  1`] = `
<div
  className="iKCvIz">
  <h3
    className="iArjKn">
    Family Member
    <button
      className="btn btn-link"
      disabled={false}
      onClick={[Function]}
      type="button">
      <span
        className="glyphicon glyphicon-pencil" />
       Edit
    </button>
  </h3>
  <div
    className="dgEelt" />
  <div
    className="row">
    <div
      className="col-md-12">
      <p
        className="cEiVmq">
        No family member is included for this test.
      </p>
    </div>
  </div>
</div>
`;
